keras/engine/topology.py

```diff
@@ -2774,7 +2774,8 @@ def save_weights_to_hdf5_group(f, layers):
         weight_values = K.batch_get_value(symbolic_weights)
         weight_names = []
         for i, (w, val) in enumerate(zip(symbolic_weights, weight_values)):
-            if hasattr(w, 'name') and w.name:
+            # cntk does not gurantee the parameter name is unique, so prefer the second approach here.
+            if hasattr(w, 'name') and w.name and K.backend() != 'cntk':
```

Comment:
Maybe some misunderstanding, the original purpose is to use autogenerated names for CNTK while keep the TF and Theano behavior unchanged.....  But after discussion, we decide use the same naming strategy as Theano for cntk, to make it more compatible. Has update the implementation.